js/main.js

```diff
@@ -1,2 +1,29 @@
 /* global data */
 /* exported data */
+
+var title = document.querySelector('#title');
+var photoUrl = document.querySelector('#photo-url');
+var notes = document.querySelector('#notes');
+var img = document.querySelector('img');
+var form = document.querySelector('form');
+
+function saveEntry(event) {
+  event.preventDefault();
+  var entry = {};
+  entry.title = title.value;
+  entry.photoUrl = photoUrl.value;
+  entry.notes = notes.value;
+  entry.id = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.unshift(entry);
+  img.setAttribute('src', 'images/placeholder-image-square.jpg');
+  form.reset();
+  localStorage.setItem('data', JSON.stringify(data));
```

Comment:
You do not need to update `localStorage` after every entry is submitted, you should just do it right before the page unloads using a `beforeunload` event listener. Additionally, all `localStorage` actions should be done in the `data.js` file so that your data model is in the same file as your code for `localStorage`.